File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        Map<Character, Integer> map  = new LinkedHashMap<>();
        for(int i = 0;i<s.length();i++){
            char c = s.charAt(i);
            map.put(c, map.getOrDefault(c,0)+1);
        }
        char a = 0;
        for (Map.Entry<Character,Integer> entry:map.entrySet()){
            if (entry.getValue().equals(1)){
                a = entry.getKey();
                break;
            }
        }
        for (int i = 0;i<s.length();i++){
            if (s.charAt(i)==a){
                return i;
            }
        }
        return -1;
    }
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        a = {}
        for i in s:
            a[i] = a.get(i,0)+1
        x=''
        # print(a)
        for i in a:
            if a[i]==1:
                x=i
                break
        # print(x)
        for b in range(len(s)):
            if s[b]==x:
                return b
        return -1